src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs

```diff
@@ -195,6 +195,49 @@ public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart,
         defenderStunMultiplier = 1f;
     }
 
+    // This method should be adjuste once totalenergy uses our strike model to adjust the result when the strike is very bad.
+    public override bool DecideCrushedThrough(
+        Agent attackerAgent,
+        Agent defenderAgent,
+        float totalAttackEnergy,
+        Agent.UsageDirection attackDirection,
+        StrikeType strikeType,
+        WeaponComponentData defendItem,
+        bool isPassiveUsage)
+    {
+        EquipmentIndex wieldedItemIndex = attackerAgent.GetWieldedItemIndex(Agent.HandIndex.OffHand);
+        if (wieldedItemIndex == EquipmentIndex.None)
+        {
+            wieldedItemIndex = attackerAgent.GetWieldedItemIndex(Agent.HandIndex.MainHand);
+        }
+
+        var weaponComponentData = wieldedItemIndex != EquipmentIndex.None
```

Comment:
weaponComponentData -> attackItem
to be consistent with the defendItem variable